java/src/main/java/ai/rapids/cudf/ParquetColumnWriterOptions.java

```diff
@@ -0,0 +1,400 @@
+  // https://github.com/rapidsai/cudf/pull/7461/commits/5ce33b40abb87cc7b76b5efeb0a3a0215f9ef6fb
+  // but it was reverted later on here:
+  // https://github.com/rapidsai/cudf/pull/7461/commits/f248eb7265de995a95f998d46d897fb0ae47f53e
+  static ParquetColumnWriterOptions DUMMY_CHILD = new ParquetColumnWriterOptions("DUMMY");
+
+  public static abstract class NestedBuilder<T extends NestedBuilder, V extends ParquetColumnWriterOptions> {
+    protected List<ParquetColumnWriterOptions> children = new ArrayList<>();
+    protected boolean isNullable = true;
+    protected String name = "";
+
+    /**
+     * Builder specific to build a Struct meta
+     */
+    protected NestedBuilder(String name, boolean isNullable) {
+      this.name = name;
+      this.isNullable = isNullable;
+    }
+
+    protected NestedBuilder() {}
+
+    protected ParquetColumnWriterOptions withColumn(String name, boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, isNullable);
+    }
+
+    protected ParquetColumnWriterOptions withDecimal(String name, int precision,
+                                                     boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, false, precision, isNullable);
+    }
+
+    protected ParquetColumnWriterOptions withTimestamp(String name, boolean isInt96,
+                                                       boolean isNullable) {
+      return new ParquetColumnWriterOptions(name, isInt96, 0, isNullable);
+    }
+
+    /**
+     * Set the list column meta.
+     * Lists should have only one child in ColumnVector, but the metadata expects a
+     * LIST column to have two children and the first child to be the
+     * {@link ParquetColumnWriterOptions#DUMMY_CHILD}.
+     * This is the current behavior in cudf and will change in future
+     * @return this for chaining.
+     */
+    public T withListColumn(ParquetListColumnWriterOptions child) {
+      assert (child.getChildColumnOptions().length == 2) : "Lists can only have two children";
+      if (child.getChildColumnOptions()[0] != DUMMY_CHILD) {
+        throw new IllegalArgumentException("First child in the list has to be DUMMY_CHILD");
+      }
+      if (child.getChildColumnOptions()[1].getColumName().isEmpty()) {
+        throw new IllegalArgumentException("Column name can't be empty");
+      }
+      children.add(child);
+      return (T) this;
+    }
+
+    /**
+     * Set a child struct meta data
+     * @return this for chaining.
+     */
+    public T withStructColumn(ParquetStructColumnWriterOptions child) {
+      for (ParquetColumnWriterOptions opt: child.getChildColumnOptions()) {
+        if (opt.getColumName().isEmpty()) {
+          throw new IllegalArgumentException("Column name can't be empty");
+        }
+      }
+      children.add(child);
+      return (T) this;
+    }
+
+    /**
+     * Set column name
+     */
+    public T withNonNullableColumn(String... name) {
```

Comment:
Should this be `withNullableColumns` so it's clearer that multiple arguments are supported? The name implies singular.  This comment applies to other places where varargs are supported.